framerail/src/lib/server/load/user.ts

```diff
@@ -0,0 +1,68 @@
+import defaults from "$lib/defaults"
+import { translate } from "$lib/server/deepwell/translate"
+import { userView } from "$lib/server/deepwell/user.ts"
+import { error, redirect } from "@sveltejs/kit"
+import { parse } from "accept-language-parser"
+
+export async function loadUser(username?: string, request, cookies) {
+  const url = new URL(request.url)
+  const domain = url.hostname
+  const sessionToken = cookies.get("wikijump_token")
+  const language = request.headers.get("Accept-Language")
+  let locales = parse(language).map((lang) =>
+    lang.region ? `${lang.code}-${lang.region}` : lang.code
+  )
+
+  if (!locales.includes(defaults.fallbackLocale)) locales.push(defaults.fallbackLocale)
+
+  const response = await userView(domain, sessionToken, locales, username)
+
+  let translateKeys: Record<string, Record<string, string | number> | {}> = {
+    ...defaults.translateKeys
+  }
+
+  const viewData = response.data
+  viewData.view = response.type
+
+  let errorStatus = null
+
+  switch (response.type) {
+    case "user_found":
+      break
+    case "user_missing":
+      viewData.user = null
+      errorStatus = 404
+      break
+    case "site_missing":
+      errorStatus = 404
+  }
+
+  if (errorStatus === null && username && viewData.user.slug !== username) {
+    throw redirect(308, `/-/user/${viewData.user.slug}`)
+  }
+
+  if (errorStatus !== null) {
+    translateKeys = Object.assign(translateKeys, {
+      "user-not-exist": {}
+    })
```

Comment:
I think spread notation is preferred now? `{ ...translateKeys, 'new-key-1': {}, 'new-key-2': {} }`